src/Microsoft.OData.Core/UriParser/Binders/FunctionCallBinder.cs

```diff
@@ -426,6 +436,52 @@ private bool TryBindIdentifier(string identifier, IEnumerable<FunctionParameterT
             return true;
         }
 
+        /// <summary>
+        /// Binds a <see cref="DottedIdentifierToken"/> for the 'isof' and 'cast' function calls.
+        /// </summary>
+        /// <param name="dottedIdentifierToken">The dotted identifier token to bind.</param>
+        /// <returns>A <see cref="QueryNode"/> representing the bound single resource node.</returns>
+        /// <exception cref="ODataException">Thrown when the token cannot be bound as a single resource node.</exception>
+        private QueryNode TryBindDottedIdentifierForIsOfOrCastFunctionCall(DottedIdentifierToken dottedIdentifierToken)
+        {
+            QueryNode parent = null;
+            IEdmType parentType = null;
+
+            if (state.ImplicitRangeVariable != null)
+            {
+                if (dottedIdentifierToken.NextToken == null)
+                {
+                    parent = NodeFactory.CreateRangeVariableReferenceNode(state.ImplicitRangeVariable);
+                    parentType = state.ImplicitRangeVariable.TypeReference.Definition;
+                }
+                else
+                {
+                    parent = this.bindMethod(dottedIdentifierToken.NextToken);
+                    parentType = parent.GetEdmType();
+                }
+            }
+
+            IEdmSchemaType childType = UriEdmHelpers.FindTypeFromModel(state.Model, dottedIdentifierToken.Identifier, this.Resolver);
+            IEdmStructuredType childStructuredType = childType as IEdmStructuredType;
+
+            if (childStructuredType == null)
+            {
+                return this.bindMethod(dottedIdentifierToken);
+            }

```

Comment:
```suggestion

            if ( childType is IEdmStructuredType childStructuredType)
            {
                return this.bindMethod(dottedIdentifierToken);
            }

```
nit: Use is operator instead

Reply:
Thanks for this. Implemented